core/src/impl/Kokkos_SharedAlloc.hpp

```diff
@@ -229,6 +244,58 @@ class SharedAllocationRecord<void, void> {
       const SharedAllocationRecord* const root, const bool detail);
 };
 
+template <class MemorySpace>
+class SharedAllocationRecordCommon : public SharedAllocationRecord<void, void> {
+ private:
+  using derived_t     = SharedAllocationRecord<MemorySpace, void>;
+  using record_base_t = SharedAllocationRecord<void, void>;
+  derived_t& self() { return *static_cast<derived_t*>(this); }
+  derived_t const& self() const { return *static_cast<derived_t const*>(this); }
+
+ protected:
+  using record_base_t::record_base_t;
+
+  void _fill_host_accessible_header_info(SharedAllocationHeader& arg_header,
```

Comment:
I thought we had settled for `impl_` prefix

Reply:
This isn't public. `impl_` is for things that need to be public for some reason (usually because friendship with some other library is difficult or impossible) but aren't part of the supported stable public API. Nothing in this class is part of the public API, but there are parts that are only used in this class hierarchy. A common convention for this is to prefix with an underscore, and since we don't really have a uniform convention for this in Kokkos, I just used that here. If you can point to somewhere else where we have a uniform convention for protected/private methods of implementation classes, I'll use that instead; otherwise, I don't see a reason to change this.

Reply:
I don't buy your argument.
![How standards proliferate](https://xkcd.com/927/)